src/test/java/io/fabric8/maven/docker/util/AuthConfigFactoryTest.java

```diff
@@ -532,13 +578,45 @@ private Server create(String id, String user, String password, String email) {
         };
     }
 
-    private void verifyAuthConfig(AuthConfig config, String username, String password, String email) {
+    private void givenEcsMetadataService(String containerCredentialsUri, String accessKeyId, String secretAccessKey, String sessionToken) throws IOException {
+        httpServer = ServerBootstrap.bootstrap()
+                .registerHandler("*", (request, response, context) -> {
+                    System.out.println("REQUEST: " + request.getRequestLine());
+                    if (containerCredentialsUri.matches(request.getRequestLine().getUri())) {
+                        response.setEntity(new StringEntity(gsonBuilder.create().toJson(ImmutableMap.of(
+                                "AccessKeyId", accessKeyId,
+                                "SecretAccessKey", secretAccessKey,
+                                "Token", sessionToken
+                        ))));
+                    } else {
+                        response.setStatusCode(SC_NOT_FOUND);
+                    }
+                })
+                .create();
+        httpServer.start();
+    }
+
+    private void setupEcsMetadataConfiguration(HttpServer httpServer, String containerCredentialsUri) {
+        environmentVariables.set("ECS_METADATA_ENDPOINT", "http://" +
+                httpServer.getInetAddress().getHostAddress()+":" + httpServer.getLocalPort());
+        environmentVariables.set("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI", containerCredentialsUri);
+    }
+
+    private void verifyAuthConfig(AuthConfig config, String username, String password, String email, String auth) {
+        assertNotNull(config);
         JsonObject params = gsonBuilder.create().fromJson(new String(Base64.decodeBase64(config.toHeaderValue().getBytes())), JsonObject.class);
-        assertEquals(username,params.get("username").getAsString());
-        assertEquals(password,params.get("password").getAsString());
+        assertEquals(username, params.get("username").getAsString());
```

Comment:
we can switch over to the AuthCofig constants if the other PR is merged.

Reply:
Done-